Changed styles for question feedback

Imran92 opened the pull request:
Resolves https://github.com/Automattic/sensei/issues/7115

## Proposed Changes
* Updated answer feedback styles for quizzes.

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Checkout this branch of pro https://github.com/Automattic/sensei-pro/pull/2421
2. Checkout this branch of course theme https://github.com/Automattic/themes/pull/7373
3. Create a course with a lesson with a quiz that has all types of questions
4. Take the quiz as a student, answer some wrong and some right
5. Grade the quiz as a teacher, provide additional feedbacks for some and don't provide for some. Mark some as wrong and some as right.
6. Check the graded quiz from the frontend.
7. Check for all course theme variations and also a couple of top themes apart from course

![Screenshot 2023-09-15 at 9 58 32 PM](https://github.com/Automattic/sensei/assets/6820724/c29795ef-a34c-4d76-8c5f-0a7d678659a5)


## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [ ] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Code is tested on the minimum supported PHP and WordPress versions
- [x] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

Imran92 committed:
Add font family

Imran92 committed:
Change icon and color of feedback

Imran92 committed:
Fix padding to match design

Imran92 committed:
Remove font size cap

Imran92 committed:
Fix position and size of icon

Imran92 committed:
Remove font weight for received grade in feedback

Imran92 committed:
Update padding in feedback content

Imran92 committed:
Remove element to stop making feedback bold

Imran92 committed:
Increase border width to match design

Imran92 committed:
Update color and border color

Imran92 committed:
Add changelog

Imran92 committed:
Merge branch 'trunk' into add/changed-styles-for-question-feedback

donnapep commented:
I've done some testing on the Course theme default variation, and here's what I found:

- It doesn't appear as though the answer feedback text (Correct / Incorrect, points, feedback description) is using the same color as the designs. The design uses `#1E1E1E` (same color is used across all themes / variations).
- The background color of "Incorrect" should be `#F2DDBE` (colors differs across themes / variations; same for background color of "Correct").
- The color of the checkmark also changes depending on the theme / variation. In the design it's `#4A7556`, but in this PR it's `#5c936b`. I'd be OK with it being the same as the text color if it's easier though.
- The whitespace below the feedback header isn't uniform:

![Screenshot 2023-09-25 at 3 13 24 PM](https://github.com/Automattic/sensei/assets/1190420/593636fe-6bfd-4508-8ab7-958021cad93e)

![Screenshot 2023-09-25 at 3 14 42 PM](https://github.com/Automattic/sensei/assets/1190420/ff15cab4-8a02-4a57-bd2a-c0fc1d577a40)

In case you need it, [here is the design I tested against](https://href.li/?https://www.figma.com/file/8mY8ui4zYK1Ylw2yIFi9Lx/?node-id=20022458%3A70849). I haven't tested other themes, but please do give those a quick check to ensure the colors match the design.

Imran92 committed:
Merge branch 'trunk' into add/changed-styles-for-question-feedback

Imran92 committed:
Change text color inside feedback element

Imran92 committed:
Update font size of feedback

Imran92 committed:
Remove margin from top of inner content in note

Imran92 commented:
> It doesn't appear as though the answer feedback text (Correct / Incorrect, points, feedback description) is using the same color as the designs. The design uses #1E1E1E (same color is used across all themes / variations).

Thanks! Updated it here https://github.com/Automattic/sensei/pull/7171/commits/60fe86aaa9278eab794a5b8057511f59e9908e85

> The background color of "Incorrect" should be #F2DDBE (colors differs across themes / variations; same for background color of "Correct").

You're right. Here's the special cases I've found
- The background color for 'incorrect' is different only for default variation and gold variation of course
- The background color for 'correct' is different only for gold variation
- The border color for 'correct' is different only for gold variation
- The border color for 'incorrect' is different only for gold variation

I've updated those in the theme PR https://github.com/Automattic/themes/pull/7373

> The color of the checkmark also changes depending on the theme / variation. In the design it's #4A7556, but in this PR it's #5c936b. I'd be OK with it being the same as the text color if it's easier though.

The icon color appears to be the same as the border color in most designs, as far as I've checked it is only slightly different in the default variation for the correct feedback. As the only place to add CSS for it is in the `theme.json`, I've intentionally left it to add less CSS in that file.

> The whitespace below the feedback header isn't uniform:

Fixed it here https://github.com/Automattic/sensei/pull/7171/commits/a0abcb7f6555fe3bf1fb63c043f4c96c5565b1e5

Thanks!

donnapep commented on assets/blocks/quiz/question-block/question.scss:
```diff
@@ -120,18 +120,23 @@ $question-input-border-color: var(--wp--preset--color--foreground, #155E65 );
 .sensei-lms-question__answer-feedback {
 	--color: #DCDCDE;
 	$horizontal-padding: 24px;
```
I believe this is orphaned code now, so we should remove it.

Imran92 replied:
Right, removed it here https://github.com/Automattic/sensei/pull/7171/commits/696ee723976d8a1a002c0faac99223592481d751

donnapep commented on assets/blocks/quiz/question-block/question.scss:
```diff
@@ -120,18 +120,23 @@ $question-input-border-color: var(--wp--preset--color--foreground, #155E65 );
-	border: 1px solid var(--color);
+	border: 2px solid var(--border-color);
 	border-radius: 1px;
+	font-family: var(--wp--preset--font-family--system);
```
This CSS variable is only available for block themes, and only those that explicitly have a custom font named `system`? We should explicitly fall back to `--wp--preset--font-family--body-font`, which it's currently defaulting to anyway.

Imran92 replied:
I've added the explicit fallback here https://github.com/Automattic/sensei/pull/7171/commits/272a9e18549e9cabe52d9256d2744be670f77804. Thanks!

donnapep reviewed (commented):
It looks like Divi has some spacing issues:

![divi - incorrect](https://github.com/Automattic/sensei/assets/1190420/30d36d09-8b91-4a3e-99a1-8bec1d496a9f)

![divi - correct](https://github.com/Automattic/sensei/assets/1190420/f5fe5c71-3146-47f3-97f9-81589361169f)

donnapep committed:
Merge branch 'trunk' into add/changed-styles-for-question-feedback

Imran92 committed:
Fix spacings in divi for question feedback

Imran92 committed:
Remove horizontal padding

Imran92 committed:
Add fallback font

Imran92 committed:
Update divi style overrides

Imran92 commented:
> It looks like Divi has some spacing issues:
> 
> ![divi - incorrect](https://user-images.githubusercontent.com/1190420/271595413-30d36d09-8b91-4a3e-99a1-8bec1d496a9f.jpg)
> 
> ![divi - correct](https://user-images.githubusercontent.com/1190420/271595427-f5fe5c71-3146-47f3-97f9-81589361169f.jpg)

Thanks @donnapep, looks like it was due to this https://github.com/Automattic/sensei/blob/2e9b39b91a0e71ea7c8f2db04e240971ba0e3f41/assets/css/3rd-party/themes/divi/learning-mode.scss#L12-L14, I'm a bit worried about this being very less specific and being applied on everything inside. But I'm not very sure how it's going to affect if removed. I've added some styles here for fixing the issues we were facing https://github.com/Automattic/sensei/pull/7171/commits/4199e49e7eaabe4d08907462ab664b3e7344eb5f
